hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -66,24 +58,13 @@
  * If the metadata table does not exist, RPC calls are used to retrieve file listings from the file system.
  * No updates are applied to the table and it is not synced.
  */
-public class HoodieBackedTableMetadata implements HoodieTableMetadata {
+public class HoodieBackedTableMetadata extends AbstractHoodieTableMetadata {
```

Comment:
Given the only subclass of `AbstractHoodieTableMetadata` is this class and the abstract class already assumes the existence of the metadata table, I am not sure if the splitting is needed per se. 

I was under the impression that we would like to implement the timeline merging as another subclass. If we stick with the new scanner approach (which I actually have grown to like actually), we no longer need the abstract class right?

Reply:
There still needs to be a place where the instant scanner result is merged with the metadata table result. I'm not sure I'm able to see how we would like this logic at scanner level. Since this should be common behavior regardless of metadata storage this is why I thought to create this

Reply:
okay that also makes sense. Personally, I would not have done the split, until a second kind of metadata impl shows up. but udit added the FileSytemBackedMetadata, it does not need the merging per se.. Still we can keep it like this for now.